hoot-core/src/main/cpp/hoot/core/conflate/DuplicateWayRemover.cpp

```diff
@@ -163,14 +179,34 @@ bool DuplicateWayRemover::_isCandidateWay(const ConstWayPtr& w) const
   return result;
 }
 
-void DuplicateWayRemover::_removeDuplicateNodes(shared_ptr<Way> w1, shared_ptr<Way> w2)
+void DuplicateWayRemover::_updateWayNameTags(shared_ptr<Way> way1, shared_ptr<Way> way2)
 {
+  //merge name tags for these two ways, if they aren't already identical
+  double namesScore = 0.0;
+  double weight = 0.0;
+  TagComparator::getInstance().compareNames(way1->getTags(), way2->getTags(), namesScore, weight);
```

Comment:
I'm not sure compareNames will do what you want it to do. The name scoring is more lenient than you want. How about?

```
// conceptual only
way1->getTags().getNames().sort() != way2->getTags().getNames().sort()
```